seissolxdmf/seissolxdmf/seissolxdmf.py

```diff
@@ -238,7 +238,7 @@ def Read1dData(self, dataName, nElements, isInt=False):
             myData = self.ReadHdf5DatasetChunk(path + filename, hdf5var, 0, MemDimension)
         else:
             filename = dataLocation
-            myData = self.ReadSimpleBinaryFile(path + dataLocation, nElements, data_prec, isInt)
+            myData = self.ReadSimpleBinaryFile(path + dataLocation, nElements, data_prec, isInt, 0)
```

Comment:
Quick question: what does the 0 do here?

Reply:
Good question!
It turns out that there is a bug in the xdmfwriter of seissol (not the python module!), as partition or fault-tags are padded, but the size written in the xdmf file in the unpadded one.
Here is an example of output:
```
    </Geometry>
    <Time Value="0"/>
    <Attribute Name="partition" Center="Cell">
     <DataItem  NumberType="Int" Precision="4" Format="Binary" Dimensions="4368">tpv5-fault_cell/mesh0/partition.bin</DataItem>
    </Attribute>
    <Attribute Name="fault-tag" Center="Cell">
     <DataItem  NumberType="Int" Precision="4" Format="Binary" Dimensions="4368">tpv5-fault_cell/mesh0/fault-tag.bin</DataItem>
    </Attribute>
    <Attribute Name="SRs" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="4368">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 4368</DataItem>
      <DataItem NumberType="Float" Precision="4" Format="Binary" Dimensions="1 2097152">tpv5-fault_cell/mesh0/SRs.bin</DataItem>
     </DataItem>
    </Attribute>
    <Attribute Name="SRd" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="4368">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 4368</DataItem>
      <DataItem NumberType="Float" Precision="4" Format="Binary" Dimensions="1 2097152">tpv5-fault_cell/mesh0/SRd.bin</DataItem>
     </DataItem>
    </Attribute>
```
Therefore if we read with idt=-1, we read a much larger array than expected (and the inferred ndt is very large in the case above and myData cannot be reshaped).
As a work-around, I set idt=0 so that just the first chunk of size nElements get read.

Reply:
I've added a comment (but it would also make sense to fix the xdmfwriter)

Reply:
https://github.com/TUM-I5/XdmfWriter/issues/13